vsomeip-sys/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+mod cxx_bridge;
+
+use autocxx::prelude::*; // use all the main autocxx functions
+
+// using autocxx to generate Rust bindings for all these
+include_cpp! {
+    #include "vsomeip/vsomeip.hpp"
+    #include "runtime_wrapper.h"
+    #include "application_wrapper.h"
+    #include "message_wrapper.h"
+    #include "payload_wrapper.h"
+    safety!(unsafe) // see details of unsafety policies described in the 'safety' section of the book
+    generate!("vsomeip_v3::runtime")
+    generate!("vsomeip_v3::application")
+    generate!("vsomeip_v3::message_base")
+    generate!("vsomeip_v3::message_t")
+    generate!("vsomeip_v3::subscription_status_handler_t")
+    generate!("vsomeip_v3::ANY_MAJOR")
+    generate!("vsomeip_v3::ANY_MINOR")
+    generate!("vsomeip_v3::ANY_INSTANCE")
+    generate!("vsomeip_v3::ANY_SERVICE")
+    generate!("vsomeip_v3::ANY_METHOD")
+    generate!("vsomeip_v3::ANY_EVENT")
+    generate!("vsomeip_v3::ANY_EVENTGROUP")
+    generate!("glue::RuntimeWrapper")
+    generate!("glue::make_runtime_wrapper")
+    generate!("glue::ApplicationWrapper")
+    generate!("glue::make_application_wrapper")
+    generate!("glue::MessageWrapper")
+    generate!("glue::make_message_wrapper")
+    generate!("glue::upcast")
+    generate!("glue::PayloadWrapper")
+    generate!("glue::make_payload_wrapper")
+    generate!("glue::set_payload_raw")
+    generate!("glue::get_payload_raw")
+    generate!("glue::create_payload_wrapper")
+}
+
+pub mod vsomeip {
+    pub use crate::ffi::vsomeip_v3::*;
+}
+
+pub mod extern_callback_wrappers;
+pub mod safe_glue;
```

Comment:
having a public API in a module called `glue` and `safe_glue` feels a little odd. Do you really need to expose these modules in their entirety? Maybe you can simple reexport some types of rename the modules?

Reply:
Generally speaking... I should probably revisit the module setup.

I agree with you. I may open an issue for this instead of addressing on this PR since it'll take a bit of rearrangement.